packages/create-schemas/src/utils.ts

```diff
@@ -0,0 +1,69 @@
+import { isAbsolute, join } from "node:path";
+import { fileURLToPath, pathToFileURL } from "node:url";
+
+function isFileURLAsText(input: string | URL): input is `file://${string}` {
+    return typeof input === "string" && input.startsWith("file://");
+}
+
+function isHttpURLAsText(
+    input: string
+): input is `http://${string}` | `https://${string}` {
+    if (!URL.canParse(input)) {
+        return false;
+    }
+
+    const url = new URL(input);
+
+    return url.protocol === "http:" || url.protocol === "https:";
+}
+
+/**
+ * Converts a path into a fully qualified path URL.
+ *
+ * - `foo` -> `file:///path/to/cwd/foo`
+ * - `https://example.com/foo` -> `https://example.com/foo`
+ *
+ * This format is works for both file path and HTTP URLs, but the result needs
+ * to be converted to a `URL` instance or into a file path depending on the
+ * usage.
+ */
+export function toFullyQualifiedURL(
```

Comment:
some thoughts, if an user provides a misformed URL/path or with trailing `/`, what is the expected behavior? Do we fail at the config loading step and notify users? I'm guessing we shouldn't try to sanitize any urls for the user right and just fail with an error there

Reply:
URLs currently aren't validated. I'm not sure what "good" URL validation would look like. I'm afraid of restricting inputs more than is necessary.

URLs/paths with a trailing `/` are technically absolute paths, and are treated as such. I wouldn't bother treating them differently since in POSIX you might want to pass an absolute path like: `/home/tjosepo/my-project/schema.json`. I was consider treating paths with a trailing `/` as relative, but I realized it might not make sense for a tool like this.

I haven't tested URL sanitization, but I believe the `pathToFileURL` function takes care of it.